File: java/src/proxy/gumball/GumballMonitorTestDrive.java
package proxy.gumball;

import java.rmi.Naming;
 
public class GumballMonitorTestDrive {
 
	public static void main(String[] args) {
		String location;
		if (args.length > 0){
            location = "rmi://" + args[0] + "/gumballmachine";
       
            GumballMonitor monitor=null;
			try {
           		GumballMachineRemote machine = 
						(GumballMachineRemote) Naming.lookup(location);
           		monitor = new GumballMonitor(machine);
				System.out.println(monitor);
        	} catch (Exception e) {
            	e.printStackTrace();
        	}
			monitor.report();
		}
	}
}

File: src/proxy/gumball/GumballMonitorTestDrive.kt
package proxy.gumball

import java.rmi.Naming


fun main(args: Array<String>) {
    val location: String
    if (args.size > 0) {
        location = "rmi://" + args[0] + "/gumballmachine"

        var monitor: GumballMonitor? = null
        try {
            val machine =
                Naming.lookup(location) as GumballMachineRemote
            monitor = GumballMonitor(machine)
            println(monitor)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        monitor!!.report()
    }
}
